lib/ClangImporter/ImportDecl.cpp

```diff
@@ -8794,23 +8796,40 @@ ClangImporter::Implementation::importSwiftAttrAttributes(Decl *MappedDecl) {
     // FIXME: Hard-code @MainActor and @UIActor, because we don't have a
     // point at which to do name lookup for imported entities.
     if (isMainActorAttr(swiftAttr)) {
-      if (SeenMainActorAttr) {
+      if (seenMainActorAttr) {
         // Cannot add main actor annotation twice. We'll keep the first
         // one and raise a warning about the duplicate.
         HeaderLoc attrLoc(swiftAttr->getLocation());
         diagnose(attrLoc, diag::import_multiple_mainactor_attr,
                  swiftAttr->getAttribute(),
-                 SeenMainActorAttr.getValue()->getAttribute());
+                 seenMainActorAttr.getValue()->getAttribute());
         continue;
       }
 
       if (Type mainActorType = SwiftContext.getMainActorType()) {
         auto typeExpr = TypeExpr::createImplicit(mainActorType, SwiftContext);
         auto attr = CustomAttr::create(SwiftContext, SourceLoc(), typeExpr);
         MappedDecl->getAttrs().add(attr);
-        SeenMainActorAttr = swiftAttr;
+        seenMainActorAttr = swiftAttr;
+      }
+
+      continue;
+    }
+
+    if (isMutabilityAttr(swiftAttr)) {
```

Comment:
I think we could also emit a warning here if `nonmutating` is applied to a non-const method: in that case `nonmutating` would be ignored, which I think is correct, however it might be hard to notice for an API vendor who forgot to add the `const` keyword to a method.

Reply:
I agree :) For completeness I also added a warning for when there aren't any mutable fields. In that case `nonmutating` is also ignored.